third_party/move/move-compiler-v2/src/pipeline/dead_store_elimination.rs

```diff
@@ -1,57 +1,202 @@
 // Copyright © Aptos Foundation
 // SPDX-License-Identifier: Apache-2.0
 
-//! Implements the "dead store elimination" transformation. This transformation pairs well with
-//! copy propagation transformation, as it removes the dead stores that copy propagation may introduce.
+//! Implements the "dead store elimination" transformation.
 //!
-//! prerequisite: the `LiveVarAnnotation` should already be computed by running the `LiveVarAnalysisProcessor`.
+//! This transformation should be run after the variable coalescing transformation,
+//! as it removes the dead stores that variable coalescing may introduce.
+//!
+//! prerequisite: the `LiveVarAnnotation` should already be computed by running the
+//! `LiveVarAnalysisProcessor` in the `track_all_usages` mode.
 //! side effect: all annotations will be removed from the function target annotations.
 //!
-//! Given live variables at each program point, this transformation removes dead stores, i.e.,
-//! assignments and loads to locals which are not live afterwards.
+//! Given live variables and all their usages at each program point,
+//! this transformation removes dead stores, i.e., assignments and loads to locals which
+//! are not live afterwards (or are live only in dead code, making them effectively dead).
 //! In addition, it also removes self-assignments, i.e., assignments of the form `x = x`.
 
 use crate::pipeline::livevar_analysis_processor::LiveVarAnnotation;
 use move_binary_format::file_format::CodeOffset;
-use move_model::model::FunctionEnv;
+use move_model::{ast::TempIndex, model::FunctionEnv};
 use move_stackless_bytecode::{
     function_target::{FunctionData, FunctionTarget},
     function_target_pipeline::{FunctionTargetProcessor, FunctionTargetsHolder},
     stackless_bytecode::Bytecode,
 };
+use std::collections::{BTreeMap, BTreeSet};
+
+/// A definition-use graph, where:
```

Comment:
Oddly, I feel like "def-use graph" (or "DefUse Graph") is more meaningful than "definition-use graph".  Using the nonstandard terminology makes me wonder how this is different.